File: Java/src/BFS_DFS/네트워크.java
package BFS_DFS;

import java.util.LinkedList;

public class 네트워크 {
    public int solution(int n, int[][] computers) {
        int answer = 0;
        boolean[] visit = new boolean[n];
        for(int i = 0; i < n; i++){
            if(visit[i]) continue;
            answer++;
            bfs(i,computers,visit);
        }
        return answer;
    }
    
    void bfs(int point, int[][] computers, boolean[] visit) {
        LinkedList<Integer> que = new LinkedList<>();
        que.add(point);
        
        while(!que.isEmpty()) {
            int next_point = que.pop();
            for(int i = 0; i < computers[next_point].length; i++) {
                if(computers[next_point][i] == 0 || visit[i]) continue;
                
                visit[i] = true;
                que.add(i);
            }
        }
    }
}

File: Python/level_3/네트워크.py
def solution(n, computers):
    answer = 0
    visit = [0 for _ in range(n)]

    def dfs(start):
        stack = []
        stack.append(start)
        visit[start] = 1

        while stack:
            start = stack.pop()
            for i in range(len(computers[start])):
                if start == i or not computers[start][i]:
                    continue
                if visit[i]:
                    continue
                visit[i] = 1
                stack.append(i)

    for i in range(n):
        if visit[i]:
            continue
        dfs(i)
        answer += 1
    return answer

n = 3
computers = [[1,1,0],[1,1,0],[0,0,1]]

print(solution(n,computers))